Failed to build on Debian Testing

Hi 
I am having some problem building gadgetron on Debian Testing, I can't even get pass the cmake process, the last few lines of cmake output is as follow:
```
...
CMake Error at chroot/CMakeLists.txt:31 (get_filename_component):
  get_filename_component unknown component -lpthread


CPACK_PACKAGING_INSTALL_PREFIX: /usr/local
Found CPack generators: DEB
Configuring incomplete, errors occurred!
See also "/home/zhitaoli/workspace/Git/gadgetron/build/release/CMakeFiles/CMakeOutput.log".
```
I cannot get anything useful from the log file, thus I am putting this here to see if any of you guys know what is going on.

On another note, I think Debian Testing is migrating to GCC 5.0, meaning some major libraries like boost provided by debian is also going to be built using gcc 5.0, as far as I know CUDA is not compatible with gcc 5.0(even the newest version 7.5), I am not sure how this is going to impact gadgetron, but I am already experiencing problem building ISMRMrd with older gcc versions(4.8, 4.9), due to libboost been compiled with gcc 5.0. So the situation is to use the system libs like boost, gcc 5.0 is required, but then to have CUDA gcc 4.8 is required, thus making boost(and maybe some other libs too) and CUDA mutually exclusive...

Two other options:
1) use the docker image (or vm) approach
https://github.com/gadgetron/gadgetron/wiki/Using-Docker <https://github.com/gadgetron/gadgetron/wiki/Using-Docker>
you can log into the docker image and use the full development environment.

2) use anaconda to create a native self-contained environment:
http://conda.pydata.org/miniconda.html <http://conda.pydata.org/miniconda.html>
I recently started setting this up here.  See this for more info:
https://github.com/conda/conda-recipes <https://github.com/conda/conda-recipes>
https://github.com/gadgetron/gadgetron-conda-recipes <https://github.com/gadgetron/gadgetron-conda-recipes>
and the beginnings of a channel with many of the bits here:
https://anaconda.org/inati <https://anaconda.org/inati>



> On Oct 10, 2015, at 10:17 AM, PlutoniumHeart <notifications@github.com> wrote:
> 
> Hi 
> I am having some problem building gadgetron on Debian Testing, I can't even get pass the cmake process, the last few lines of cmake output is as follow:
> 
> ...
> CMake Error at chroot/CMakeLists.txt:31 (get_filename_component):
>   get_filename_component unknown component -lpthread
> 
> 
> CPACK_PACKAGING_INSTALL_PREFIX: /usr/local
> Found CPack generators: DEB
> Configuring incomplete, errors occurred!
> See also "/home/zhitaoli/workspace/Git/gadgetron/build/release/CMakeFiles/CMakeOutput.log".
> I cannot get anything useful from the log file, thus I am putting this here to see if any of you guys know what is going on.
> 
> On another note, I think Debian Testing is migrating to GCC 5.0, meaning some major libraries like boost provided by debian is also going to be built using gcc 5.0, as far as I know CUDA is not compatible with gcc 5.0(even the newest version 7.5), I am not sure how this is going to impact gadgetron, but I am already experiencing problem building ISMRMrd with older gcc versions(4.8, 4.9), due to libboost been compiled with gcc 5.0. So the situation is to use the system libs like boost, gcc 5.0 is required, but then to have CUDA gcc 4.8 is required, thus making boost(and maybe some other libs too) and CUDA mutually exclusive...
> 
> —
> Reply to this email directly or view it on GitHub <https://github.com/gadgetron/gadgetron/issues/227>.
>

Well, being on testing, this is the sort of thing that happens from time to time. Once Ubuntu 15.10 (which will probably have GCC 5) is out, we will probably start having a look at this. As long as CUDA is not supported with GCC 5, we will not be switching. But I'm sure it will be in the next few months (maybe CUDA 7.5, we will see).

It's inconvenient, but the cost of being at the bleeding edge.

Our reference platform is Ubuntu LTS (currently 14.04) if you rely on the Gadgetron for your stuff, that's the platform to be on. But again, it will work with GCC 5, when all the dependencies settle down and we start looking at it.

------
Sent from a mobile device - please excuse brevity and/or typos.

> On Oct 10, 2015, at 10:17 AM, PlutoniumHeart <notifications@github.com> wrote:
> 
> Hi 
> I am having some problem building gadgetron on Debian Testing, I can't even get pass the cmake process, the last few lines of cmake output is as follow:
> 
> ...
> CMake Error at chroot/CMakeLists.txt:31 (get_filename_component):
>   get_filename_component unknown component -lpthread
> 
> 
> CPACK_PACKAGING_INSTALL_PREFIX: /usr/local
> Found CPack generators: DEB
> Configuring incomplete, errors occurred!
> See also "/home/zhitaoli/workspace/Git/gadgetron/build/release/CMakeFiles/CMakeOutput.log".
> I cannot get anything useful from the log file, thus I am putting this here to see if any of you guys know what is going on.
> 
> On another note, I think Debian Testing is migrating to GCC 5.0, meaning some major libraries like boost provided by debian is also going to be built using gcc 5.0, as far as I know CUDA is not compatible with gcc 5.0(even the newest version 7.5), I am not sure how this is going to impact gadgetron, but I am already experiencing problem building ISMRMrd with older gcc versions(4.8, 4.9), due to libboost been compiled with gcc 5.0. So the situation is to use the system libs like boost, gcc 5.0 is required, but then to have CUDA gcc 4.8 is required, thus making boost(and maybe some other libs too) and CUDA mutually exclusive...
> 
> —
> Reply to this email directly or view it on GitHub.
>

--- a/cmake/FindCUDA_advanced.cmake
+++ b/cmake/FindCUDA_advanced.cmake
@@ -1,3 +1,5 @@
+set(CUDA_USE_STATIC_CUDA_RUNTIME OFF CACHE BOOL "")
+
 find_package(CUDA 5.5)
 
 # Check for GPUs present and their compute capability

